src/Common/src/TypeSystem/IL/HelperExtensions.cs

```diff
@@ -34,37 +34,20 @@ public static MethodDesc GetStringInitializer(this MethodDesc constructorMethod)
             MethodSignature sig = new MethodSignature(
                 MethodSignatureFlags.Static, 0, constructorMethod.OwningType, parameters);
 
-            MethodDesc result = constructorMethod.OwningType.GetMethod("Ctor", sig);
-
-            // TODO: Better exception type. Should be: "CoreLib doesn't have a required thing in it".
-            if (result == null)
-                throw new NotImplementedException();
-
+            MethodDesc result = constructorMethod.OwningType.GetKnownMethod("Ctor", sig);
             return result;
         }
 
         public static MetadataType GetHelperType(this TypeSystemContext context, string name)
         {
-            MetadataType helperType = context.SystemModule.GetType("Internal.Runtime.CompilerHelpers", name, false);
-            if (helperType == null)
-            {
-                // TODO: throw the exception that means 'Core Library doesn't have a required thing in it'
-                throw new NotImplementedException();
-            }
-
+            MetadataType helperType = context.SystemModule.GetKnownType("Internal.Runtime.CompilerHelpers", name);
```

Comment:
They force people to put stuff into the Internal.Runtime.CompilerHelpers namespace. I think it's goodness.